lib/genesis/helpers.go

```diff
@@ -636,6 +802,10 @@ func addAuthoritiesValues(k1, k2 string, kt crypto.KeyType, value []byte, gen *G
 		return err
 	}
 
-	gen.Genesis.Runtime[k1][k2] = authAddrs
+	if k1 == "Babe" {
+		gen.Genesis.Runtime.Babe.Authorities = authAddrs
+	} else if k1 == "Grandpa" {
+		gen.Genesis.Runtime.Grandpa.Authorities = authAddrs
+	}
```

Comment:
use a switch instead + use lowercase babe and grandpa as that's what you pass in callers.

```suggestion
	switch k1 {
	case "babe":
		gen.Genesis.Runtime.Babe.Authorities = authAddrs
	case "grandpa":
		gen.Genesis.Runtime.Grandpa.Authorities = authAddrs
	}
```